### PR TITLE
Alignment semantic documentation vertical

### PR DESCRIPTION
**Pull request by apinzenoehler**
proposed reviewers:

@larswik 
@czapplitec 
@jmirtsch 

Note: this pull request was separated from alignment documentation rework for better GitHub handling of reviewing and merging.

### CODE REVIEW

**Comment by SergejMuhic** [comment]
Not sure if @jmirtsch can make it. I might take his place.

**Review by larswik — COMMENTED** [review]
Very clear and understandable! Just a few comments to check.

File: IFC4x3/Constants/c/CIRCULARARC_2oOPMXkJz5HPDm05vV_d6O.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="CIRCULARARC_2oOPMXkJz5HPDm05vV_d6O" Name="CIRCULARARC" UniqueId="b26195a1-b93f-4545-9370-005e5ffa7198">
+	<Documentation>Vertical alignment segment where the derivative of vertical angle with respect to sloping length along the track (3D length) is constant.
```
[the remark this answers is not in the document]
Comment (SergejMuhic) [reply IFC4x3/Constants/c/CIRCULARARC_2oOPMXkJz5HPDm05vV_d6O.xml:3]: Agree with both.

File: IFC4x3/Constants/p/PARABOLICARC_0I31$UIXXDOuYw__sJmCDk.xml [hunk 2]
```diff
@@ -1,5 +1,18 @@
 <?xml version="1.0" encoding="utf-8"?>
 <DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PARABOLICARC_0I31UIXXDOuYw__sJmCDk" Name="PARABOLICARC" UniqueId="120c1fde-4a18-4d63-88ba-fbed93c0c36e">
-	<Documentation>Parabolic arc type of segment.</Documentation>
+	<Documentation>Vertical alignment segment where the derivative of gradient with respect to chainage is constant.
```
[the remark this answers is not in the document]
Comment (SergejMuhic) [reply IFC4x3/Constants/p/PARABOLICARC_0I31$UIXXDOuYw__sJmCDk.xml:3]: Could also use the "arc length".

**Review by SergejMuhic — COMMENTED** [review]
Looks good. @larswik comments I agree with.

File: IFC4x3/Constants/c/CIRCULARARC_2oOPMXkJz5HPDm05vV_d6O.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (apinzenoehler) [reply IFC4x3/Constants/c/CIRCULARARC_2oOPMXkJz5HPDm05vV_d6O.xml:3]: Valid points.

The idea was to connect the wording in the documentation directly with EN 13803 where the terms are coming from.

Maybe we can put the EN13803 original into a separate paragraph. I would really would like to keep EN13803 in the text, it was the first text I found which made this clear distinction between 3D-length (sloping length) and 2D-length (chainage). This in turn gives us a clear reference point for an unified IFC rule for application of vertical alignment.

File: IFC4x3/Constants/c/CLOTHOID_35dYjaqAz3VPYqYea78nAG.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="CLOTHOID_35dYjaqAz3VPYqYea78nAG" Name="CLOTHOID" UniqueId="c59e2b64-d0af-437d-98b4-8a8907231290">
+	<Documentation>Vertical alignment segment where the derivative of vertical angle with respect to sloping length along the track (3D length) obeys a linear change.
```
[the remark this answers is not in the document]
Comment (apinzenoehler) [reply IFC4x3/Constants/c/CLOTHOID_35dYjaqAz3VPYqYea78nAG.xml:3]: Here we do not have the EN13803 reference. The vertical clothoid is currently only known from a Chinese highspeed regulation.

File: IFC4x3/Constants/p/PARABOLICARC_0I31$UIXXDOuYw__sJmCDk.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (apinzenoehler) [reply IFC4x3/Constants/p/PARABOLICARC_0I31$UIXXDOuYw__sJmCDk.xml:3]: EN13803 clause 3.5:
chainage: longitudinal distance along the horizontal projection of the track centre line.

In the case of the PARABOLICARC it is therefore the horizontal length and not the "arc length".

BTW: Quote from "Ellis British Railway Engineering Encyclopaedia", 2n edition, p72:

**Chainage**

a) "The location along the track measured from the previous whole Milepost in Chains (22 Yards, 20.1169 metres)
b) The absolute position along the line of a Route measured from some designated reference point in a standard measurement unit, not necessarily Chains. See also Metreage.

Note: I have big sympathy for avoiding the term "chainage".

File: IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricConstraintResource/Types/IfcAlignmentVerticalSegmentTypeEnum/Documentation.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+The IfcAlignmentVerticalSegmentTypeEnum indicates the type of a segment of a vertical alignment segment (IfcAlignmentVerticalSegment).
+
+
+| Vertical curvature | Segmenttype        | Enumeration Values |
+|:----|:------------------|:----------|
+| No vertical curvature | constant gradient        | CONSTANTGRADIENT |
+| Derivative of gradient with respect to chainage is constant | Vertical curve, parabola  | PARABOLICARC |
+| Derivative of vertical angle with respect to sloping length along the track is constant | Vertical curve, circular | CIRCULARARC  |
```
[the remark this answers is not in the document]
Comment (apinzenoehler) [reply IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometricConstraintResource/Types/IfcAlignmentVerticalSegmentTypeEnum/Documentation.md:8]: Would fit. 

Same arguments concerning EN13803 as above apply.

File: IFC4x3/Constants/p/PARABOLICARC_0I31$UIXXDOuYw__sJmCDk.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (SergejMuhic) [reply IFC4x3/Constants/p/PARABOLICARC_0I31$UIXXDOuYw__sJmCDk.xml:3]:
```suggestion
	<Documentation>Vertical alignment segment where the derivative of gradient with respect to distance along is constant.
```

**Pull request merged**